daemon/src/position_metrics.rs

```diff
@@ -168,10 +168,20 @@ impl Cfd {
                 is_open: true,
                 ..self
             },
-            ContractSetupFailed => Self {
-                is_failed: true,
-                ..self
-            },
+            ContractSetupFailed => {
+                // This is needed because we have a CFD (bd6221b0-47f6-4b14-ac9b-04cfa355cb9b) in
+                // the database where `OfferRejected` and `ContractSetupFailed` was appended. This
+                // was due to a bug that has since been fixed; `OfferRejected` and
+                // `ContractSetupFailed` are mutually exclusive.
```

Comment:
🙃 I think this comment should exist without any reference to production. The motivation is to ensure that we only have one state at a time, in case we can emit more than one "final" event.

Reply:
> 🙃 I think this comment should exist without any reference to production. The motivation is to ensure that we only have one state at a time, in case we can emit more than one "final" event.

I agree! But before tackling this I'd like to take a step back and look at the big picture of aggregates currently in the code-base. We had the idea of crating one aggregate for projection and metrics at some point. I am not sure if it's a good idea but I'd like to evaluate it once we have a look into how to make the projection "better".
I feel the aggregates have been sources of bugs for some times and we keep running into similar/same problems with different aggregates when we change things. For me this points to not being the right abstraction. 
I will merge this fix for now. So we don't forget the refactoring idea I recorded it as a ticket but would not action it until we have a better high level picture :)
https://github.com/itchysats/itchysats/issues/2022